File: backend/DTOs/JobApplicationResponseDto.cs
using Apollo_hire.API.Models.Enums;

namespace Apollo_hire.API.DTOs
{
    public class JobApplicationResponseDto
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        
        public string CompanyName { get; set; } = null!;
        public string JobTitle { get; set; } = null!;
        public string Location { get; set; } = null!;
        public WorkType WorkType { get; set; }
        public ApplicationStatus Status { get; set; }
        public string Source { get; set; } = null!;
        public string? SourceUrl { get; set; }
        public DateTime? AppliedDate { get; set; }
        public string? Notes { get; set; }
        
        public bool IsArchived { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        
        public int EventCount { get; set; }
    }
}

File: frontend/src/app/job-applications/models/job-application-response-dto.ts
import { ApplicationStatus } from "./application-status";
import { WorkType } from "./work-type";

export interface JobApplicationResponseDto {
  id: string;
  userId: string;
  companyName: string;
  jobTitle: string;
  location: string;
  workType: WorkType;
  status: ApplicationStatus;
  source: string;
  sourceUrl?: string | null;
  appliedDate?: string | null;
  notes?: string | null;
  isArchived: boolean;
  createdAt: string;
  updatedAt: string;
  eventCount: number;
}
